### PR TITLE
Cherry-picking from main

### PR DESCRIPTION
**Pull request by rauletorresc**
**Context:** We need some commits from main.

**Description of the Change:** Cherry-pick them.

### CODE REVIEW

**Comment by erick-xanadu** [comment]
@dime10 should #1113 be included as well for Qrisp?

**Comment by dime10** [comment]
> @dime10 should #1113 be included as well for Qrisp?

That would nice 👍

**Comment by dime10** [comment]
Also any any bug fixes we have merged, we might as well release those (unless conflicts are getting too complex).

**Comment by rauletorresc** [comment]
We are adding ZNE stuff only for this release

File: frontend/test/pytest/test_mitigation.py [hunk 1]
```diff
@@ -52,17 +54,21 @@ def circuit(x):
...
 
 
 @pytest.mark.parametrize("params", [0.1, 0.2, 0.3, 0.4, 0.5])
 @pytest.mark.parametrize("extrapolation", [quadratic_extrapolation, exponential_extrapolate])
-def test_multiple_measurements(params, extrapolation):
+@pytest.mark.parametrize("folding", ["global", "local-all"])
+def test_multiple_measurements(params, extrapolation, folding):
     """Test that without noise the same results are returned for multiple measurements"""
-    skip_if_exponential_extrapolation_unstable(params, extrapolation)
+    skip_if_exponential_extrapolation_unstable(params, extrapolation, threshold=0.4)
```
Comment (rmoyard) [inline frontend/test/pytest/test_mitigation.py:71]:
```suggestion
    skip_if_exponential_extrapolation_unstable(params, extrapolation, threshold=0.5)
```

File: doc/releases/changelog-0.8.1.md [hunk 2]
```diff
@@ -2,14 +2,59 @@
 
 <h3>New features</h3>
 
-<h3>Improvements</h3>
+* Zero-Noise Extrapolation (ZNE) local folding: Introduces the option to fold gates locally as well
```
Comment (rmoyard) [inline doc/releases/changelog-0.8.1.md:5]: @josh146 Can you take a look at the changelog?

**Review by rmoyard — COMMENTED** [review]
Looks good, can check that all builds are passing?

**Comment by rauletorresc** [comment]
> Looks good, can check that all builds are passing?

I activated Build Wheels and CPL rc/rc/rc, let's see

File: doc/releases/changelog-0.8.1.md (hunk 2 above)
Comment (josh146) [reply doc/releases/changelog-0.8.1.md:5]: Will do!

File: doc/releases/changelog-0.8.1.md [hunk 3]
```diff
@@ -2,14 +2,84 @@
...
+  definition. It now needs to be a list of positive odd integers, as we don't support the fractional
+  part.
+  [(#1120)](https://github.com/PennyLaneAI/catalyst/pull/1120)
 
 <h3>Contributors</h3>
 
 This release contains contributions from (in alphabetical order):
+
+Joey Carter,
+Alessandro Cosentino,
```
Comment (josh146) [inline doc/releases/changelog-0.8.1.md:81]: I've added @cosenal here as well, as I saw he made contributions/suggestions on #1006!

**Review by josh146 — APPROVED** [review]
thanks!

File: doc/releases/changelog-0.8.1.md (hunk 3 above)
Comment (rauletorresc) [reply doc/releases/changelog-0.8.1.md:87]: Cool!

**Review by rmoyard — APPROVED** [review]
Thanks, don't forget to update the version in `_version.py`

File: doc/releases/changelog-0.8.1.md (hunk 2 above)
Comment (rauletorresc) [reply doc/releases/changelog-0.8.1.md]: Done

**Comment by rauletorresc** [comment]
> Thanks, don't forget to update the version in `_version.py`

Done!

File: doc/releases/changelog-0.8.1.md (hunk 3 above)
Comment (cosenal) [reply doc/releases/changelog-0.8.1.md:87]: Thanks, @josh146!

**Pull request merged**